ltsp/server/ipxe/55-ipxe.sh

```diff
@@ -28,46 +37,66 @@ ipxe_cmdline() {
 
-    # Prepare the menu text for all images and chroot
+    # Prepare the menu text for all images and chroots
     key=0
-    items=""
-    gotos=":images"
-    img_name=$(re list_img_names -i)
-    set -- $img_name
-    for img_name in "$@"; do
-        key=$((key+1))
-        title=$(echo_values "$(ipxe_name "$img_name.img")")
-        title=${title:-$img_name.img}
-        items="${items:+"$items\n"}$(printf "item --key %d %-20s %s" "$((key%10))" "$img_name" "$title")"
-        gotos=":$img_name\n$gotos"
-    done
-    r_items=""
-    r_gotos=":roots"
-    img_name=$(re list_img_names -c)
-    set -- $img_name
-    for img_name in "$@"; do
+    items=""   # The menu items to be inserted.
+    gotos=""   # Jump targets setting correct ${img} var for menu items.
+    methods="" # Jump targets with the respective boot_method.
+
+    # INFO: The images/chroots automatism has been removed. It has been replaced
+    # by the BOOT_METHOD functionality, which can achieve the same thing but
+    # without hardwiring LTSP to any specific BOOT_METHOD.
+    # As a side effect some code repititions have also been remove.
+
+    # Extracts all vars with pattern BOOT_METHOD_[0-9]* and prepares boot
+    # options for their respective dirs.
+    boot_method_vars=$(echo_vars "BOOT_METHOD_[0-9]*")
+    set -- $boot_method_vars
+
+    for boot_method_var in "$@"; do
+      eval "subdir=\$${boot_method_var}_DIR"
+      eval "boot_method=\$${boot_method_var}"
+      eval "boot_method_name=\$${boot_method_var}_NAME"
+      boot_method_target=$(ipxe_lowercase ${boot_method_var})
+      methods="${methods}${boot_method_target}:\nset cmdline_boot_method $boot_method \&\& goto ltsp\n"
+      items="${items:+"$items\n"}$(printf "item --gap %s" "$boot_method_name")"
+      boot_names=$(re list_boot_names $BASE_DIR/$subdir)
+      set -- $boot_names
+      for boot_name in "$@"; do
         key=$((key+1))
-        title=$(echo_values "$(ipxe_name "$img_name")")
-        title=${title:-$img_name}
-        r_items="${r_items:+"$r_items\n"}$(printf "item --key %d %-20s %s" "$((key%10))" "r_$img_name" "$title")"
-        r_gotos=":r_$img_name\nset img $img_name \&\& goto roots\n$r_gotos"
+        [ $(ipxe_prepvar "${boot_method_var}_${boot_name}") = "${BOOT_DEFAULT}" ] && is_default=" --default" || is_default=""
+        item_label=${key}_${boot_name}
+        title=$(echo_values "$(ipxe_prepvar "${boot_method_var}_${boot_name}")")
+        title="${title:-${boot_name} (./${subdir}${subdir:+/}${boot_name})}"
+        if [ $key -lt 10 ]; then
+          items="${items:+"$items\n"}$(printf "item%s --key %d %-20s %s" "$is_default" "$((key%10))" "$item_label" "$title")"
+        else
+          items="${items:+"$items\n"}$(printf "item%s %-28s %s" "$is_default" "$((key%10))" "$item_label" "$title")"
+        fi
+        gotos="$gotos:${item_label}:\nset img $boot_name \&\& goto ${boot_method_target}\n"
```

Comment:
Same here - one ":" to much:

Should be:
gotos="$gotos:${item_label}\nset img $boot_name \&\& goto ${boot_method_target}\n"

Patch coming shortly.